esp/services/ws_ecl/ws_ecl_service.cpp

```diff
@@ -214,9 +214,9 @@ bool CWsEclService::init(const char * name, const char * type, IPropertyTree * c
             daliAddress.append(*daliServers++);
     }
 
-    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
-    Owned<IConstEnvironment> environment = factory->openEnvironment();
-    Owned<IPropertyTree> pRoot = &environment->getPTree();
+    Owned<IPropertyTree> pRoot = getEnvironmentPTreeWithUpdate();
+    if (!pRoot)
+        throw MakeStringException(-1, "Failed to get environment information.");
```

Comment:
See comment above about accessing raw meta info.
I think this is a different change from the issue raised in the JIRA + should be tackled not by exposing the meta info more directly, but by extending the IConstEnvironment interfaces where appropriate to access the info. you need.

For example here, I think you're walking RoxieCluster's .. if there isn't already methods to give you a list of Roxie's and/or not a good way to get to their instance addresses, the environment interfaces should be extended.

Should open a separate JIRA for that though.